server/chat-plugins/thecafe.tsx

```diff
@@ -176,22 +186,20 @@ export const commands: Chat.ChatCommands = {
 export const pages: Chat.PageTable = {
 	foodfight(query, user, connection) {
 		if (!user.named) return Rooms.RETRY_AFTER_LOGIN;
-		let buf = `|title|Foodfight\n|pagehtml|<div class="pad ladder"><h2>Foodfight Dish list</h2>`;
+		this.title = 'Foodfight';
 		const room = Rooms.get('thecafe');
 		if (!room) return this.errorReply(`Room not found.`);
-		if (!user.can('mute', null, room)) {
-			return buf + `<p>Access denied</p></div>`;
-		}
+		this.checkCan('mute', null, room);
 		const content = Object.values(dishes).map(
-			([dish, ...ingredients]) => `<tr><td>${dish}</td><td>${ingredients.join(', ')}</td></tr>`
+			([dish, ...ingredients]) => <tr><td>{dish}</td><td>{ingredients.join(', ')}</td></tr>
 		).join('');
 
-		if (!content) {
-			buf += `<p>There are no dishes in the database.</p>`;
-		} else {
-			buf += `<table><tr><th><h3>Dishes</h3></th><th><h3>Ingredients</h3></th></tr>${content}</table>`;
-		}
-		buf += `</div>`;
-		return buf;
+		return <div class="pad ladder">
+			<h2>Foodfight Dish list</h2>
+				{content ?
+					<table><tr><th><h3>Dishes</h3></th><th><h3>Ingredients</h3></th></tr>{content}</table> :
+					<p>There are no dishes in the database.</p>
+				}
+			</div>
```

Comment:
```suggestion
		return <div class="pad ladder">
			<h2>Foodfight Dish list</h2>
			{content ?
				<table><tr><th><h3>Dishes</h3></th><th><h3>Ingredients</h3></th></tr>{content}</table> :
				<p>There are no dishes in the database.</p>
			}
		</div>;
```

This clearly isn't checked by eslint; that'll need to be fixed, too.